src/decorators/legacy/legacy-decorator-utils.ts

```diff
@@ -0,0 +1,44 @@
+export type PropertyOrGetterDescriptor = (
```

Comment:
Maybe just call this file `decorator-utils`? Since it's already in the legacy folder